### PR TITLE
Add lie_learn

### PR DESCRIPTION
**Pull request by mariogeiger**
@conda-forge/help-python-c

### CODE REVIEW

**Comment by tadeu** [comment]
From checking the jobs output it looks like `numpy` should be added as a dependency:

```
    Running command python setup.py egg_info
    Running setup.py (path:/tmp/pip-req-build-1_h75oru/setup.py) egg_info for package from file://$SRC_DIR
    ERROR: Could not find a version that satisfies the requirement numpy (from versions: none)
    ERROR: No matching distribution found for numpy
```

**Comment by tadeu** [comment]
Now it looks like it also needs `scipy <1.3`:

```
    Installed c:\miniconda\conda-bld\lie_learn_1581970027449\_h_env\lib\site-packages\lie_learn-0.0.0-py2.7-win-amd64.egg
    Processing dependencies for lie-learn==0.0.0
    Searching for scipy<1.3
```

```
    Installed /home/conda/staged-recipes/build_artifacts/lie_learn_1581969853856/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/lib/python2.7/site-packages/lie_learn-0.0.0-py2.7-linux-x86_64.egg
    Processing dependencies for lie-learn==0.0.0
    Searching for scipy<1.3
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-req-build-oJ0QP0/setup.py", line 82, in <module>
        ':python_version>="3.0"': ['scipy', 'numpy'],
```

**Comment by tadeu** [comment]
You can also add a `skip` for Python 2 if you won't use that, to help you focus on Python 3 requirements

**Comment by mariogeiger** [comment]
> Now it looks like it also needs `scipy <1.3`

How do you see that?
Is it by running `.circleci/run_docker_build.sh`?

**Comment by tadeu** [comment]
Cool, it looks like `scipy` was the last missing requirement, the important builds are passing! It's almost good to go, the recipe just need some tests now, as commented by the linter in this message: https://github.com/conda-forge/staged-recipes/pull/10861#issuecomment-586650383

Just some import tests would be fine, but you might also want to run unit tests if you'd like, take a look here: https://conda-forge.readthedocs.io/en/latest/testing.html#testing-importing

**Comment by tadeu** [comment]
Suggestion:

```diff
test:
+  imports:
+    - lie_learn
  requires:
    - pytest
  source_files:
    - lie_learn
+  commands:
+    - pytest --pyargs lie_learn
```

File: recipes/lie_learn/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    - pip
+    - cython
+    - requests
+    - numpy
+    - scipy
+  run:
+    - python
+    - cython
+    - requests
+    - numpy
```
Comment (marcelotrevisani) [inline recipes/lie_learn/meta.yaml]:
```suggestion
    - {{ pin_compatible('numpy') }}
```

File: recipes/lie_learn/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    - scipy
+
+test:
+  requires:
+    - pytest
+  source_files:
+    - lie_learn
```
Comment (marcelotrevisani) [inline recipes/lie_learn/meta.yaml]: you are not testing anything here, could you please add
```suggestion
    - lie_learn
  imports:
    - lie_learn
```

**Comment by marcelotrevisani** [comment]
> Suggestion:
> 
> ```diff
> test:
> +  imports:
> +    - lie_learn
>   requires:
>     - pytest
>   source_files:
>     - lie_learn
> +  commands:
> +    - pytest --pyargs lie_learn
> ```

Just saw your comment after I did mine, sorry 
:s

**Comment by mariogeiger** [comment]
Thanks a lot for the help.

@marcelotrevisani it looks like your change affecting `numpy` dependency installs a too old version of numpy [TypeError: load() got an unexpected keyword argument 'encoding'](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=123404&view=logs&j=10bda42a-d420-5c28-a43f-bc4e66405e5c&t=bef775d0-a1a2-55a3-9721-17147029baf0&l=881)

**Comment by tadeu** [comment]
@mariogeiger, you should probably pin a minimum required version of `numpy` now, check the "Notes" section at: https://conda-forge.org/docs/maintainer/knowledge_base.html#building-against-numpy

File: recipes/lie_learn/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  version: {{ version }}
+
+source:
+  url: https://github.com/AMLab-Amsterdam/lie_learn/archive/v{{ version }}.tar.gz
+  sha256: 820937f84f5e505c00fb54e52ab357be72d5998481de8144af33fc4254d5b670
+
+build:
+  number: 0
+  skip: True  # [py<35]
+  script: "{{ PYTHON }} -m pip install . -vv && cd .. && {{ PYTHON }} -c \"from lie_learn.representations.SO3.wigner_d import wigner_D_matrix; wigner_D_matrix(2, 1, 2, 3)\""
```
Comment (marcelotrevisani) [inline recipes/lie_learn/meta.yaml:15]:
```suggestion
  script: "{{ PYTHON }} -m pip install . -vv
```

**Pull request closed without merging**